wine-quality-ml_implementation_speculative/src/main/scala/pipelines/examples/modelserving/speculative/model/WineDecider.scala

```diff
@@ -0,0 +1,36 @@
+package pipelines.examples.modelserving.speculative.model
+
+import com.lightbend.modelserving.model.ModelResultMetadata
+import com.lightbend.modelserving.model.speculative.Decider
+import pipelines.examples.modelserving.winequality.data.{ WineRecord, WineResult }
+import pipelines.examples.modelserving.winequality.result.ModelDoubleResult
+import util.control.Breaks._
+
+class WineDecider extends Decider[WineResult] {
+
+  private val random = new scala.util.Random
+
+  // The simplest decider returning the last result
+  override def decideResult(results: List[WineResult]): WineResult = {
+
+    results.size match {
+      case 0 ⇒ // No results, can only happen if we timed out
```

Comment:
Sorry, whats wrong with this?

Reply:
You are returning a bogus record. If I wrote a production application with this logic, would it be acceptable to return a bogus, made-up record and also not log an error? I don't think it would be acceptable. That's why this function needs to return `Either` and the error case needs to handled higher in the stack.

Reply:
Wait, decider is doing the right thing. If there is no results, the only thing that it can do is to return an error that everything timed out, which it does. If all results are errors, this error is returned, and only if we have at least one result, then it will calculate result

Reply:
There is no higher level thing - the decider is the master